### PR TITLE
Adds US Privacy API generation

### PR DESCRIPTION
**Pull request by banderson623**
Building on PR #4 this adds support for the [US Privacy API as outlined by the IAB](https://github.com/InteractiveAdvertisingBureau/USPrivacy/blob/master/CCPA/USP%20API.md) (not a PDF!) This is an important component for programmatic advertising, google ad manager and Prebid understanding the user's preferences.

**Exposing a new Generator `installPrivacyAPI`**
```js
PrivacyCompliance.Generator.installPrivacyAPI();
```

Which creates the `window.__uspapi()` function:

> `__uspapi(Command, Version, Callback)`

```js
__uspapi('getUSPData', 1 , (uspData, success) => {
  if(success) {
    console.log(uspData);
  } else {
    // do something else
  }
});
```
resulting in the following being logged to the console

```js
{
  version: 1,
  uspString: '1YNN'
}
```

It also offers support for [iframe messaging](https://github.com/InteractiveAdvertisingBureau/USPrivacy/blob/master/CCPA/USP%20API.md#without-safeframes-using-postmessage) 

> To locate an ancestor frame capable of responding to postMessage() API calls search for an ancestor frame that has a child frame named '__uspapiLocator'.

So it creates an iframe with `<iframe name="__uspapiLocator"> ... </iframe>` at the top level to proxy messages across iframes.

### CODE REVIEW

File: src/frameworks/us_privacy_string_and_api_generator.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+ * for more information on the US Privacy string see:
+ * https://github.com/InteractiveAdvertisingBureau/USPrivacy/blob/master/CCPA/US%20Privacy%20String.md#us-privacy-string-format
+ */
+class UsPrivacyStringAndAPIGenerator extends FrameworkBase {
+  constructor() {
+    super();
+    this.window = window;
+    this.document = document;
+  }
+  supportedGenerators() {
```
Comment (andrew-a-dev) [inline src/frameworks/us_privacy_string_and_api_generator.js:21]: (MINOR) Probably want a newline between L15 & L16

Comment (banderson623) [reply src/frameworks/us_privacy_string_and_api_generator.js:21]: doh, thank you!

File: src/frameworks/us_privacy_string_and_api_generator.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+  }
+
+  installPrivacyAPI(callback = () => {}) {
+    this.window.__uspapi = this.handleUSPrivacyAPI.bind(this);
+    this.createUspapiFrame();
+    callback(true);
+  }
+
+  // Private methods ---------
+  buildUsPrivacyString() {
```
Comment (andrew-a-dev) [inline src/frameworks/us_privacy_string_and_api_generator.js:41]: _Minor/pedantry_: Could we standardise on capitalisation for "US Privacy String" in methods/variables? (I prefer `USPrivacy`, as below, but don't really care, as long as it's consistent.)

**Review by andrew-a-dev — APPROVED** [review]
🚢  I'm still a little unclear on the role the iframe plays in all of this, but all of the code looks good to me. Nice stuff. 👍  (I left a couple minor-edit-suggestions, but definitely not blocking.)

**Pull request merged**